### PR TITLE
Adding tests For Api endpoints

### PR DESCRIPTION
**Pull request by mihrab34**
This PR adds unit and API test to the existing endpoints as specified in issue #10

### CODE REVIEW

File: doc/notes.org [hunk 1]
```diff
@@ -4,6 +4,8 @@
   source env/bin/activate
   pip install django
   pip install djangorestframework
+  pip install django-cors-headers
+  pip install djangorestframework-simplejwt
```
[the remark this answers is not in the document]
Comment (sgarciahelguera) [reply doc/notes.org:8]: checkout main, I updated the notes, (git merge main should bring that to your branch)

**Review by sgarciahelguera — CHANGES_REQUESTED** [review]
Please go over my comments.

File: cubeseed/.pylintrc [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+    invalid-name,
+    missing-docstring,
+    protected-access,
+    too-few-public-methods,
+    # handled by black
+    format,
+    # We anticipate #3512 where it will become optional
+    fixme,
+    cyclic-import,
+    import-error,
```
[the remark this answers is not in the document]
Comment (mihrab34) [reply cubeseed/.pylintrc:109]: I've seen the pylint update you made. I actually thought i removed it already. Will do that Asap

File: cubeseed/settings.py [hunk 3]
```diff
@@ -25,7 +25,7 @@
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
 
-## FIXME: this is good for development unacceptable for production
```
[the remark this answers is not in the document]
Comment (mihrab34) [reply cubeseed/settings.py:28]: okay

File: cubeseed/userauth/tests.py [hunk 4]
```diff
@@ -1,3 +1,74 @@
...
+    def test_get_user_list(self):
+        """
+        Test to get all the users in the db
+        """
+        self.test_register_user()
+
+        url = reverse('userauth/users')
+        request = self.client.get(url, format='json')
+        force_authenticate(request, user=self.user)
+        response = request + 'HTTP_AUTHORIZATION' +':'+ 'Token ' + self.user.auth_token.key
```
Comment (sgarciahelguera) [inline cubeseed/userauth/tests.py:70]: The auth header should be
```
Authorization: Bearer <token-here>
```

Comment (sgarciahelguera) [reply cubeseed/userauth/tests.py:70]: I am not familiarized with "force_authenticate" but if that is giving you  the tokens the same way userauth/token does, it should work

Comment (mihrab34) [reply cubeseed/userauth/tests.py]: I'm using the `userauth/token` now as the force_authenticate was giving me 401 error

Comment (sgarciahelguera) [reply cubeseed/userauth/tests.py]: Great, did it work?

File: .vscode/settings.json [hunk 5]
```diff
@@ -13,6 +13,9 @@
         "pylint_django.checkers.migrations",
         "--django-settings-module=cubeseed.settings",
         "--max-line-length=120",
-        "--disable=missing-docstring"
+        "--disable=missing-docstring",
```
[the remark this answers is not in the document]
Comment (sgarciahelguera) [reply .vscode/settings.json:16]: Have you seen this? 
If you agree that the missing-docstring covers all the strings please remove the others:
```
        "--disable=missing-module-docstring",
        "--disable=missing-class-docstring",
        "--disable=missing-function-docstring"
```

File: cubeseed/userauth/tests.py
Comment (sgarciahelguera) [inline cubeseed/userauth/tests.py:1]: There are some commented tests at the end

**Review by sgarciahelguera — CHANGES_REQUESTED** [review]

**Comment by sgarciahelguera** [comment]
I added a github action to run tests on Pull Requests, you should merge from main so you can see here the results of the tests running

File: cubeseed/userauth/tests.py (hunk 4 above)
Comment (mihrab34) [reply cubeseed/userauth/tests.py]: yes, it did

**Comment by mihrab34** [comment]
> I added a github action to run tests on Pull Requests, you should merge from main so you can see here the results of the tests running

File: .vscode/settings.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
     "[python]": {
-        "editor.defaultFormatter": "ms-python.black-formatter"
```
Comment (sgarciahelguera) [inline .vscode/settings.json:3]: why did you change the formatter?

**Review by sgarciahelguera — CHANGES_REQUESTED** [review]
Please go over my comments.
Also please revert changes to FIXME comments.

**Review by sgarciahelguera — APPROVED** [review]
Looks good
